docs/rest_api/users.rst

```diff
@@ -0,0 +1,541 @@
+
+.. vale off
+
+Edit User
+*********
+
+.. vale on
+
+.. code-block:: php
+
+   <?php
+
+   $id   = 1;
+   $data = array(
+       'lastName' => 'Doeboe',
+   );
+
+   // Create new a User of ID 1 isn't found?
+   $createIfNotFound = true;
+
+   $user = $userApi->edit($id, $data, $createIfNotFound);
+
+Edit a new User. User that this supports PUT or PATCH depending on the desired behavior.
+
+**PUT** creates a User if the given ID doesn't exist and clears all the User information, adds the information from the request.
+
+**PATCH** fails if the User with the given ID doesn't exist and updates the User field values with the values from the request.
+
+.. vale off
+
+**HTTP Request**
+
+.. vale on
+
+To edit a User and return a 404 if the User isn't found:
+
+``PATCH /users/ID/edit``
+
+To edit a User and create a new one if the User isn't found:
+
+``PUT /users/ID/edit``
+
+**POST Parameters**
+
+.. list-table::
+   :header-rows: 1
+
+   * - Name
+     - Type
+     - Description
+   * - ``username``
+     - string
+     - Username that's used to log into Mautic.
+   * - ``firstName``
+     - string
+     - First Name of the User
+   * - ``lastName``
+     - string
+     - Last Name of the User
+   * - ``email``
+     - string
+     - Email of the User
+   * - ``position``
+     - string
+     - User's position title
+   * - ``role``
+     - int
+     - Role ID
+   * - ``timezone``
+     - string
+     - Timezone of the User
+   * - ``signature``
+     - string
+     - Signature of the User for use in Emails
+
+**Response**
+
+If ``PUT``, the expected response code is ``200`` if editing a User or ``201`` if creating a new one.
+
+If ``PATCH``, the expected response code is ``200``.
+
+**Properties**
+
+Same as `Get User <#get-user>`_.
+
+.. vale off
+
+Delete User
+************
+
+.. vale on
+
+.. code-block:: php
+
+   <?php
+
+   $user = $userApi->delete($id);
+
+Delete a User.
+
+.. vale off
+
+**HTTP Request**
+
+.. vale on
+
+``DELETE /users/ID/delete``
+
+**Response**
+
+``Expected Response Code: 200``
+
+**Properties**
+
+Same as `Get User <#get-user>`_.
+
+.. vale off
+
+Get Self User
+*************
+
+.. vale on
+
+.. code-block:: php
+
+   <?php
+
+   $user = $userApi->getSelf();
+
+Get a self User.
+
+.. vale off
+
+**HTTP Request**
+
+.. vale on
+
+``GET /users/self``
+
+**Response**
+
+``Expected Response Code: 200``
+
+**Properties**
+
+Same as `Get User <#get-user>`_.
+
+.. vale off
+
+Check User Permissions
+**********************
+
+.. vale on
+
+.. code-block:: php
+
+   <?php
+   $permission = array('user:users:create', 'user:users:edit');
+   $user = $userApi->checkPermission($id, $permission);
+
+Get a self User.
+
+.. vale off
+
+**HTTP Request**
+
+.. vale on
+
+``GET /users/ID/permissioncheck``
+
+**Response**
+
+``Expected Response Code: 200``
+
+.. code-block:: json
+
+   {
+     "user:users:create":true,
+     "user:users:edit":true
+   }
+
+**Properties**
+
+array of requested permissions of string in case of just one
```

Comment:
```suggestion
An array of requested permissions, or a string where returning only one value.
```

I think this is what you mean but please check to be sure of the intended context!

Reply:
I'll test this endpoint in a sec and will update the description here. Thanks for noticing!